pkg/config/templates/rke2-90-harvester-server.yaml

```diff
@@ -1,7 +1,7 @@
 cni: multus,canal
-cluster-cidr: 10.52.0.0/16
-service-cidr: 10.53.0.0/16
-cluster-dns: 10.53.0.10
+cluster-cidr: {{ or .ClusterPodCIDR "10.52.0.0/16" }}
```

Comment:
the installer part lgtm. In the backend, when we promote a worker node to a server node, we also replace the config, maybe we need to tune this too:
https://github.com/harvester/harvester/blob/ef97c6c4fa49086a082c12682ef5287d4d3777f5/deploy/charts/harvester/templates/configmap.yaml#L77-L85

Reply:
@bk201 @ibrokethecloud Let me know what you think of the approach in 563845df3ceb39d58a031d6a6ce0978e60c9e1a3 and https://github.com/harvester/harvester/pull/7156. I try to inject the custom values into the `promote.sh` script defined in the `harvester-system/harvester-helpers` config map as Helm variables.

Reply:
looks good to me